packages/backend-common/src/reading/AwsS3UrlReader.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import aws, { Credentials, S3 } from 'aws-sdk';
+import {
+  ReaderFactory,
+  ReadTreeResponse,
+  SearchResponse,
+  UrlReader,
+} from './types';
+import getRawBody from 'raw-body';
+import {
+  AwsS3IntegrationConfig,
+  readAwsS3IntegrationConfig,
+} from '@backstage/integration';
+
+const AMAZON_AWS_HOST = '.amazonaws.com';
+
+const parseURL = (
+  url: string,
+): { path: string; bucket: string; region: string } => {
+  let { host, pathname } = new URL(url);
+
+  pathname = pathname.substr(1);
+
+  const validHost = new RegExp(
+    /^[a-z\d][a-z\d\.-]{1,61}[a-z\d]\.s3\.[a-z\d-]+\.amazonaws.com$/,
+  );
+  if (!validHost.test(host)) {
+    throw new Error(`not a valid AWS S3 URL: ${url}`);
+  }
+
+  const [bucket] = host.split(/\.s3\.[a-z\d-]+\.amazonaws.com/);
+  host = host.substring(bucket.length);
+  const [, , region, ,] = host.split('.');
+
+  return {
+    path: pathname,
+    bucket: bucket,
+    region: region,
+  };
+};
+
+export class AwsS3UrlReader implements UrlReader {
+  static factory: ReaderFactory = ({ config, logger }) => {
+    if (!config.has('integrations.awsS3')) {
+      return [];
+    }
+    const awsS3Config = readAwsS3IntegrationConfig(
+      config.getConfig('integrations.awsS3'),
+    );
+    let s3: S3;
+    if (!awsS3Config.accessKeyId || !awsS3Config.secretAccessKey) {
+      logger.info(
+        'awsS3 credentials not found in config. Using default credentials provider.',
```

Comment:
> `integrations.awsS3` not found in app config. AWS S3 integration will use default AWS credentials if set in environment.


Perhaps this is a bit more explanatory in the logs, as this is printed when the backend starts. I'm also questioning if it should be `logger.debug` since a huge majority of people won't have this configured, and they won't even care, to be honest.

Reply:
Changed the message to the one you suggested and changed it to `logger.debug`